src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSMessageAttributeHelper.cs

```diff
@@ -0,0 +1,82 @@
+// <copyright file="AWSMessageAttributeHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Linq;
+using Amazon.Runtime;
+using Amazon.Runtime.Internal;
+
+namespace OpenTelemetry.Contrib.Instrumentation.AWS.Implementation;
+
+internal class AWSMessageAttributeHelper
+{
+    // SQS/SNS message attributes collection size limit according to
+    // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html and
+    // https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html
+    private const int MaxMessageAttributes = 10;
+
+    private readonly IAWSMessageAttributeFormatter attributeFormatter;
+
+    internal AWSMessageAttributeHelper(IAWSMessageAttributeFormatter attributeFormatter)
+    {
+        this.attributeFormatter = attributeFormatter ?? throw new ArgumentNullException(nameof(attributeFormatter));
+    }
+
+    internal bool TryAddParameter(ParameterCollection parameters, string name, string value)
+    {
+        var index = this.GetNextAttributeIndex(parameters, name);
+        if (!index.HasValue)
+        {
+            return false;
+        }
+        else if (index >= MaxMessageAttributes)
+        {
+            // TODO: Add logging (event source).
+            return false;
+        }
+
+        var attributePrefix = this.attributeFormatter.AttributeNamePrefix + $".{index.Value}";
+        parameters.Add(attributePrefix + ".Name", name.Trim());
```

Comment:
I don't think it is safe to call `Trim` in this method. We should use the name and value as-is.

Reply:
This code is moved to implementations of `IRequestContextAdapter.AddAttribute(...)` method. Trims have been removed.